File: companion/types/plant.ts
export type CareCategory = 'watering' | 'fertilization' | 'pruning' | 'repotting';
export type PlantLocation = 'bedroom' | 'balcony' | 'office' | 'living room' | 'kitchen' | 'bathroom' | 'other';

export interface CareSchedule {
    category: CareCategory;
    frequency: number; // in days
    lastPerformed?: Date;
    notes?: string;
}

export interface Plant {
    id: string;
    name: string;
    species?: string;
    imageUri?: string;
    location?: PlantLocation;
    notes?: string;
    careSchedules: CareSchedule[];
    createdAt: Date;
    updatedAt: Date;
}

// Input types for mutations
export interface PlantCreateInput {
    name: string;
    species?: string;
    location?: PlantLocation;
    imageUri?: string;
    notes?: string;
    careSchedules: CareSchedule[];
}

export interface PlantUpdateInput {
    name?: string;
    species?: string;
    location?: PlantLocation;
    imageUri?: string;
    notes?: string;
    careSchedules?: CareSchedule[];
}

File: api/src/entities/plant.rs
pub struct Plant {
    pub id: i32,
    pub name: String,
    pub owner_id: i32,
}

pub struct PlantDb {
    pub id: i32,
    pub name: String,
    pub owner_id: i32,
}

impl From<PlantDb> for Plant {
    fn from(db: PlantDb) -> Self {
        Plant {
            id: db.id,
            name: db.name,
            owner_id: db.owner_id,
        }
    }
}
